cmd/xl-v1-multipart.go

```diff
@@ -341,8 +341,10 @@ func (xl xlObjects) PutObjectPart(ctx context.Context, bucket, object, uploadID
 
 	// Delete the temporary object part. If PutObjectPart succeeds there would be nothing to delete.
 	defer xl.deleteObject(ctx, minioMetaTmpBucket, tmpPart, writeQuorum, false)
-	if data.Size() > 0 || data.Size() == -1 {
-		if pErr := xl.prepareFile(ctx, minioMetaTmpBucket, tmpPartPath, data.Size(), onlineDisks, xlMeta.Erasure.BlockSize, xlMeta.Erasure.DataBlocks, writeQuorum); pErr != nil {
+
+	if data.Size() >= 0 {
```

Comment:
I have a small doubt here .. The objects whose size is unknown will have `data.Size() == -1` ( eg. Compressed objects ). By removing this here we omit few posix checks like [https://github.com/minio/minio/blob/9e3fce441e22ccd044bef1c1dec1d175e6d8ef19/cmd/posix.go#L260](url) in `prepareFile`. Is this intentional ?
Anyway we cant allocate disk space for such objects.. Can we have `min-disk-space` checks atleast ?

Reply:
We cannot have -1 be passed to prepareFile() -  the previous code was buggy. It's okay for compressed objects to fail since it would fail anyways. We can't do such checks for disk space anyways in a conclusive manner for compression.

Also min-disk-space is only during startup.